lib/charon/session_store/redis_store.ex

```diff
@@ -153,25 +161,38 @@ defmodule Charon.SessionStore.RedisStore do
   @impl true
   def get_all(user_id, type, config) do
     mod_conf = get_mod_config(config)
-    {user_id_str, type_str, key_prefix} = key_data(user_id, type, mod_conf)
-
-    with {:ok, keys = [_ | _]} <-
-           get_valid_session_keys(user_id_str, type_str, key_prefix, mod_conf),
-         {:ok, values} <- mod_conf.redix_module.command(["MGET" | keys]) do
-      values |> Stream.map(&deserialize(&1, mod_conf, config)) |> Enum.reject(&is_nil/1)
-    else
-      {:ok, []} -> []
-      other -> other
+    key_data = key_data(user_id, type, mod_conf)
+    session_set_key = session_set_key(key_data)
+    now = Internal.now()
+
+    with {:ok, keys} <- get_valid_session_keys(key_data, mod_conf),
+         {:ok, values} <-
+           [["HVALS", session_set_key]]
+           |> then(fn commands ->
+             case keys do
+               [] -> commands
+               _ -> [["MGET" | keys] | commands]
+             end
+           end)
+           |> mod_conf.redix_module.pipeline() do
```

Comment:
because upsert/2 deletes the old storage format, we don't deduplicate the results